example-app/src/main/java/com/adyen/checkout/example/ui/googlepay/compose/SessionsGooglePayViewModel.kt

```diff
@@ -0,0 +1,209 @@
+                countryCode = keyValueStorage.getCountry(),
+                shopperLocale = keyValueStorage.getShopperLocale(),
+                splitCardFundingSources = keyValueStorage.isSplitCardFundingSources(),
+                isExecuteThreeD = keyValueStorage.isExecuteThreeD(),
+                isThreeds2Enabled = keyValueStorage.isThreeds2Enabled(),
+                redirectUrl = savedStateHandle.get<String>(SessionsGooglePayActivity.RETURN_URL_EXTRA)
+                    ?: error("Return url should be set"),
+                shopperEmail = keyValueStorage.getShopperEmail(),
+                allowedPaymentMethods = listOf(paymentMethodType),
+                installmentOptions = getSettingsInstallmentOptionsMode(keyValueStorage.getInstallmentOptionsMode()),
+                showInstallmentAmount = keyValueStorage.isInstallmentAmountShown(),
+            ),
+        ) ?: return null
+
+        return getCheckoutSession(sessionModel, googlePayConfiguration)
+    }
+
+    private suspend fun getCheckoutSession(
+        sessionModel: SessionModel,
+        googlePayConfiguration: GooglePayConfiguration,
+    ): CheckoutSession? {
+        return when (val result = CheckoutSessionProvider.createSession(sessionModel, googlePayConfiguration)) {
+            is CheckoutSessionResult.Success -> result.checkoutSession
+            is CheckoutSessionResult.Error -> null
+        }
+    }
+
+    override fun onAvailabilityResult(isAvailable: Boolean, paymentMethod: PaymentMethod) {
+        viewModelScope.launch {
+            if (isAvailable) {
+                updateState { it.copy(uiState = SessionsGooglePayUIState.ShowButton) }
+            } else {
+                onError()
+            }
+        }
+    }
+
+    override fun onAction(action: Action) {
+        updateState { it.copy(action = action) }
+    }
+
+    override fun onError(componentError: ComponentError) {
+        Log.e(TAG, "Component error occurred")
+        onError()
+    }
+
+    override fun onFinished(result: SessionPaymentResult) {
+        updateState {
+            it.copy(uiState = SessionsGooglePayUIState.FinalResult(getFinalResultState(result)))
+        }
+    }
+
+    private fun getFinalResultState(result: SessionPaymentResult): ResultState = when (result.resultCode) {
+        "Authorised" -> ResultState.SUCCESS
+        "Pending",
+        "Received" -> ResultState.PENDING
+
+        else -> ResultState.FAILURE
+    }
+
+    private fun onError() {
+        updateState { it.copy(uiState = SessionsGooglePayUIState.FinalResult(ResultState.FAILURE)) }
+    }
+
+    private fun updateState(block: (SessionsGooglePayState) -> SessionsGooglePayState) {
+        _googlePayState.update(block)
+    }
+
+    fun availabilityChecked() {
+        updateState { it.copy(checkAvailability = null) }
+    }
+
+    fun buttonClicked() {
+        updateState { it.copy(uiState = SessionsGooglePayUIState.ShowComponent) }
+    }
+
+    fun actionConsumed() {
```

Comment:
What do you think of renaming this method to `onActionConsumed()` to keep the naming consistent. Similarly `availabilityChecked `, `buttonClicked `, `activityResultHandled`, `newIntentHandled ` could be renamed too.